### PR TITLE
Fix handling of possibly-null return from GetServiceAsync

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs [hunk 1]
```diff
@@ -62,7 +62,9 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
         private async Task InitializeAndPopulateSnippetsCacheAsync(Shell.IAsyncServiceProvider asyncServiceProvider)
         {
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
-            var textManager = (IVsTextManager2)await asyncServiceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
+            var textManager = (IVsTextManager2?)await asyncServiceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
```
Comment (Youssef1313) [inline src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs:65]: Small question:
Isn't `var` always assumes a nullable reference type? Why/when the cast can change the behavior?

Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs:65]: The problem here is entirely constrained to the right hand side. `object?` cannot be explicitly cast to `object` without a warning or a suppression.

Comment (Youssef1313) [reply src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs:65]: @sharwell Thanks a lot for clarifying

**Comment by RikkiGibson** [comment]
@sharwell I believe this needs to be backported to release/dev16.8 going by this PR: https://github.com/dotnet/roslyn/pull/47327

Seems like due to unlucky timing (right before snap) the change went into the release/dev16.8 branch.

**Comment by sharwell** [comment]
@RikkiGibson Rebased to 16.8

**Comment by RikkiGibson** [comment]
Merging as all validations we believe could be affected by this change have passed.

**Pull request merged**